test/ctia/stores/es/crud_test.clj

```diff
@@ -555,15 +571,17 @@
 
             (check-fn
              {:msg "user can only delete the data they have access to."
-              :query {:query-string "title1"
+              :query {:full-text {:query "title1"
+                                  :mode  :query_string}
                       :filter-map {:confidence "High"}}
               :matched []
               :ident ident-2
               :deleted? false})
 
             (check-fn
              {:msg "matched entities must be properly deleted"
-              :query {:query-string "title1"
+              :query {:full-text {:query "title1"
+                                  :mode  :query_string}
```

Comment:
there is a lot of boiler plate here perhaps you should factorize it with a helper:
``` clojure
(defn query-string
   ([query] (query-string query nil) 
   ([query options]
      (into {:query query
             :mode  :query_string}
             options))
```

and even derive it into a more generic helper in this [ns](https://github.com/threatgrid/ctia/blob/master/src/ctia/http/routes/common.clj):

``` clojure
(defn full-text
   ([mode query] (query-string mode query  nil) 
   ([mode query  options]
       (into {:query query}
               options)))
             
(def query-string (partial full-text :query_string))
(def simple-query-string (partial full-text :simple_query_string))
(def multi-match (partial full-text :multi_match))
```

At some point I think that we should integrate it into [ductile](https://github.com/threatgrid/ductile/blob/master/src/ductile/query.clj) to promote our Elasticsearch querying practices  :-)

Reply:
I'd rather choose not to hide it behind an abstraction and let the callers set query parameters directly on every level. One thing maybe we could do is to make `:mode` field optional and if it's not set - use `:query_string` by default. e.g.:

```
(check-fn
  {:query {:full-text {:query "title1"}
           :filter-map {:confidence "High"}}
```

Reply:
> One thing maybe we could do is to make :mode field optional and if it's not set - use :query_string by default.

Yes, I like it.